### PR TITLE
feat: refresh password

### PR DESCRIPTION
**Pull request by gurjmatharu**

### CODE REVIEW

File: services/tenant-ui/frontend/src/store/innkeeper/innkeeperTenantsStore.ts [hunk 1]
```diff
@@ -22,6 +22,7 @@ import {
 import { RESERVATION_STATUS_ROUTE } from '@/helpers/constants';
 import { API_PATH, RESERVATION_STATUSES } from '@/helpers/constants';
 import { useConfigStore } from '../configStore';
+import test from 'node:test';
```
Comment (jamshale) [inline services/tenant-ui/frontend/src/store/innkeeper/innkeeperTenantsStore.ts:25]: Looks like unused import

File: services/tenant-ui/frontend/src/store/innkeeper/innkeeperTenantsStore.ts [hunk 2]
```diff
@@ -171,6 +172,50 @@ export const useInnkeeperTenantsStore = defineStore('innkeeperTenants', () => {
...
+    console.log('> reservationStore.refreshCheckInPassword');
+    error.value = null;
+    loading.value = true;
+
+    // Don't keep this as state, make sure the password doesn't hang around in memory
+    let refreshResponse: ApproveResponse = {};
+    await acapyApi
+      .putHttp(API_PATH.INNKEEPER_RESERVATIONS_REFRESH_PASSWORD(id), payload)
+      .then((res) => {
+        refreshResponse = res.data;
```
Comment (jamshale) [inline services/tenant-ui/frontend/src/store/innkeeper/innkeeperTenantsStore.ts:189]: When there is one statement you could do this as a single line. This is fine though. I'm not sure what the project preference is. 

Personally I like doing one liners without curly braces. I think it's cleaner, but not important.

File: services/tenant-ui/frontend/src/components/innkeeper/reservation/RefreshPassword.vue [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  <Button
+    :label="$t('reservations.refreshPassword')"
+    class="p-button-rounded p-button-text"
+    icon="pi pi-refresh"
+    @click="refresh"
+  />
+</template>
+
+<script setup lang="ts">
+// Vue
```
Comment (jamshale) [inline services/tenant-ui/frontend/src/components/innkeeper/reservation/RefreshPassword.vue:11]: I feel like the import categories are getting a bit out of control. I know it's like this in other places. I started doing my imports in 2 categories. Libraries and source code. You don't have to change this I just want to mention it.

File: services/tenant-ui/frontend/src/components/innkeeper/reservation/RefreshPassword.vue [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { useInnkeeperTenantsStore } from '@/store';
+import { storeToRefs } from 'pinia';
+import InputText from 'primevue/inputtext';
+const innkeeperTenantsStore = useInnkeeperTenantsStore();
+const { loading } = storeToRefs(useInnkeeperTenantsStore());
+import { useConfigStore } from '@/store';
+import { useI18n } from 'vue-i18n';
+const { config } = storeToRefs(useConfigStore());
+const toast = useToast();
+const { t } = useI18n();
```
Comment (loneil) [inline services/tenant-ui/frontend/src/components/innkeeper/reservation/RefreshPassword.vue:27]: Would organize these a bit more, keeping the use*Store ones together and the storetorefs ones next to each other, the primevue imports next to each other etc

**Comment by jamshale** [comment]
Nice work 👍

**Comment by loneil** [comment]
Unsetting the auto-approve temporarily in pr values to test something out on this deployment

File: services/tenant-ui/frontend/src/components/innkeeper/reservation/ReservationsHistory.vue [hunk 5]
```diff
@@ -40,6 +40,12 @@
       >
         <template #body="{ data }">
           <StatusChip :status="data.state" />
+          <RefreshPassword
+            v-if="data.state === 'approved'"
+            :id="data.reservation_id"
+            :email="data.contact_email"
+            @success="showModal"
+          />
```
Comment (loneil) [inline services/tenant-ui/frontend/src/components/innkeeper/reservation/ReservationsHistory.vue:48]: Minor styling but can we align these?

image

**Review by loneil — COMMENTED** [review]
Looks good to me, tiny style change request if you can do that

**Review by loneil — APPROVED** [review]
Great job! Should be good to merge now.

For future PRs can you add a bit to the description of the PR. Short list of changes, any relevant screenshots if UI changes involved, any testing notes or future considerations, etc. Thanks!

**Pull request merged**